Fix incorrect behavior of ListViewSubItem (#4761)
dotnet/winforms: docs/accessibility/listview-expected-behavior.md

expected accessibility properties of all visible items 
of a [ListView](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview)v.

- [Overview](#Overview)
- [ListView](#ListView)
    - [Inspect tree for ListView](##Inspect-tree-for-ListView)
- [ListViewGroup](#ListViewGroup)
    - [Inspect tree for a ListView with displayed ListViewGroups](##Inspect-tree-for-a-ListView-with-displayed-ListViewGroups)
	- [Inspect tree for a ListView with disabled visual styles](##Inspect-tree-for-a-ListView-with-disabled-visual-styles)	
	- [Inspect tree for a ListView with disabled ListViewGroups](##Inspect-tree-for-a-ListView-with-disabled-ListViewGroups)
	- [Inspect tree for a ListView in virtual mode](##Inspect-tree-for-a-ListView-in-virtual-mode)
	- [Inspect tree for a ListView without ListViewGroups](##Inspect-tree-for-a-ListView-without-ListViewGroups)
- [ListViewSubItem](#ListViewSubItem)
    - [Inspect tree for a ListViewItem](##Inspect-tree-for-a-ListViewItem)
	- [Inspect tree for a ListViewItem (Details view)](##Inspect-tree-for-a-ListViewItem-(Details-view))
	- [Inspect tree for a ListViewItem (Tile view)](##Inspect-tree-for-a-ListViewItem-(Tile-view))

> [ListView](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview) is a native controls is provided by Windows. 

Reviewer: ```suggestion
[ListView](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview) is a Win32 native control provided by Windows. 
```

> After implementation [UIA](https://docs.microsoft.com/dotnet/framework/ui-automation/ui-automation-overview) support on the Winforms side the responsibility for 
> ListView accessible objects is now on Winforms. A ListView 
> accessible object and its children must have correct accessibility 
> propeties and actions to make ListView accessible for customers.

Reviewer: ```suggestion
Windows Forms runtime provides a support for [UI Automation](https://docs.microsoft.com/dotnet/framework/ui-automation/ui-automation-overview) and, thus, responsible for 
providing required accessible objects with correct properties and actions to make all aspects of the ListView control accessible for customers.
```

# ListView

## Inspect tree for ListView

> ListView accessibility tree should contain all visible items depending on its [view](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.view).

Reviewer: ```suggestion
ListView accessibility tree must contain all visible items based on the selected [`View` mode](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.view).
```

<details>
<summary>1. Details view</summary>
</br>

![listview-inspect-details-view-tree][listview-inspect-details-view-tree]

</details>
</br>

<details>
<summary>2. LargeIcon view</summary>
</br>

![listview-inspect-largeicon-view-tree][listview-inspect-largeicon-view-tree]

</details>
</br>

<details>
<summary>3. List view</summary>
</br>

![listview-inspect-list-view-tree][listview-inspect-list-view-tree]

</details>
</br>

<details>
<summary>4. SmallIcon view</summary>
</br>

![listview-inspect-smallicon-view-tree][listview-inspect-smallicon-view-tree]

</details>
</br>

<details>
<summary>5. Tile view</summary>
</br>

![listview-inspect-tile-view-tree][listview-inspect-tile-view-tree]

</details>
</br>

# ListViewGroup

> The ListView can always contain [ListViewGroup](https://docs.microsoft.com/dotnet/api/system.windows.forms.listviewgroup), but depending on the settings of the application and the ListView, these groups can be displayed or hidden.

Reviewer: ```suggestion
The ListView always contains one or more [`ListViewGroup`](https://docs.microsoft.com/dotnet/api/system.windows.forms.listviewgroup), but depending on a configuration of the operating system, application and/or the ListView itself, these groups can be visible or not.
```

## Inspect tree for a ListView with displayed ListViewGroups

> When [visual styles](https://docs.microsoft.com/dotnet/api/system.windows.forms.application.enablevisualstyles) are enabled, ListViewGroup display is enabled ([ShowGroups](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.showgroups) property), ListView has [groups](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.groups) and ListView is not in [virtual mode](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.virtualmode), then ListViewGroups are displayed for all views except "List" view. ListViewItem without a ListViewGroup are displayed in the default ListViewGroup

Reviewer: This is too complex and convoluted. Try presenting it as a list, e.g.:
```suggestion
A `ListViewGroup` is display if the following conditions are met:
1. An application has [visual styles](https://docs.microsoft.com/dotnet/api/system.windows.forms.application.enablevisualstyles) are enabled.
2. A ListView is not in a [virtual mode](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.virtualmode).
3. ...
```

> ```

Reviewer: Here and everywhere else where applicable

```suggestion
```cs
```

listView.Items.Add(new ListViewItem(new string[] { "Item 2", "SubItem 21", "SubItem 22", "SubItem 23" }, imageIndex: 0, group: listViewGroup));
listView.Items.Add(new ListViewItem(new string[] { "Item 3", "SubItem 31" }, imageIndex: 0));
```

<details>
<summary>1. Details view</summary>
</br>

![listview-withgroup-inspect-details-view-tree][listview-withgroup-inspect-details-view-tree]

</details>
</br>

<details>
<summary>2. LargeIcon view</summary>
</br>

![listview-withgroup-inspect-largeicon-view-tree][listview-withgroup-inspect-largeicon-view-tree]

</details>
</br>

<details>
<summary>3. List view</summary>
</br>

![listview-withgroup-inspect-list-view-tree][listview-withgroup-inspect-list-view-tree]

</details>
</br>

<details>
<summary>4. SmallIcon view</summary>
</br>

![listview-withgroup-inspect-smallicon-view-tree][listview-withgroup-inspect-smallicon-view-tree]

</details>
</br>

<details>
<summary>5. Tile view</summary>
</br>

![listview-withgroup-inspect-tile-view-tree][listview-withgroup-inspect-tile-view-tree]

</details>
</br>

## Inspect tree for a ListView with disabled visual styles

> We do not show ListViewGroups when visual styles are disabled

Reviewer: ```suggestion
ListViewGroups are not visible when visual styles are disabled.
```

listView.Items.Add(new ListViewItem(new string[] { "Item 2", "SubItem 21", "SubItem 22", "SubItem 23" }, imageIndex: 0, group: listViewGroup));
listView.Items.Add(new ListViewItem(new string[] { "Item 3", "SubItem 31" }, imageIndex: 0));
```

<details>
<summary>1. Details view</summary>
</br>

![listview-disabledstyles-inspect-details-view-tree][listview-disabledstyles-inspect-details-view-tree]

</details>
</br>

<details>
<summary>2. LargeIcon view</summary>
</br>

![listview-disabledstyles-inspect-largeicon-view-tree][listview-disabledstyles-inspect-largeicon-view-tree]

</details>
</br>

<details>
<summary>3. List view</summary>
</br>

![listview-disabledstyles-inspect-list-view-tree][listview-disabledstyles-inspect-list-view-tree]

</details>
</br>

<details>
<summary>4. SmallIcon view</summary>
</br>

![listview-disabledstyles-inspect-smallicon-view-tree][listview-disabledstyles-inspect-smallicon-view-tree]

</details>
</br>

<details>
<summary>5. Tile view</summary>
</br>

![listview-disabledstyles-inspect-tile-view-tree][listview-disabledstyles-inspect-tile-view-tree]

</details>
</br>

## Inspect tree for a ListView with disabled ListViewGroups

> We do not show ListViewGroups when [ShowGroups](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.showgroups) property has False value

Reviewer: ```suggestion
ListViewGroups are not visible when [`ShowGroups`](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview.showgroups) property is set to `false`.
```

The same approach below

listView.Items.Add(new ListViewItem(new string[] { "Item 2", "SubItem 21", "SubItem 22", "SubItem 23" }, imageIndex: 0));
listView.Items.Add(new ListViewItem(new string[] { "Item 3", "SubItem 31" }, imageIndex: 0));
```

<details>
<summary>1. Details view</summary>
</br>

![listview-withoutgroup-inspect-details-view-tree][listview-withoutgroup-inspect-details-view-tree]

</details>
</br>

<details>
<summary>2. LargeIcon view</summary>
</br>

![listview-withoutgroup-inspect-largeicon-view-tree][listview-withoutgroup-inspect-largeicon-view-tree]

</details>
</br>

<details>
<summary>3. List view</summary>
</br>

![listview-withoutgroup-inspect-list-view-tree][listview-withoutgroup-inspect-list-view-tree]

</details>
</br>

<details>
<summary>4. SmallIcon view</summary>
</br>

![listview-withoutgroup-inspect-smallicon-view-tree][listview-withoutgroup-inspect-smallicon-view-tree]

</details>
</br>

<details>
<summary>5. Tile view</summary>
</br>

![listview-withoutgroup-inspect-tile-view-tree][listview-withoutgroup-inspect-tile-view-tree]

</details>
</br>

# ListViewSubItem

## Inspect tree for a ListViewItem

> The [ListViewItem](https://docs.microsoft.com/dotnet/api/system.windows.forms.listviewitem) can always contain [ListViewSubItem](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.listviewitem.listviewsubitem), but depending on the settings of the application, the ListView and ListViewItem these ListViewSubItems can be displayed or hidden. in addition, ListViewSubItems are not displayed when the ListView is in "List", "SmallIcon" or "LargeIcon" views

Reviewer: Same as above

> The [ListViewItem](https://docs.microsoft.com/dotnet/api/system.windows.forms.listviewitem) can always contain [ListViewSubItem](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.listviewitem.listviewsubitem), but depending on the settings of the application, the ListView and ListViewItem these ListViewSubItems can be displayed or hidden. in addition, ListViewSubItems are not displayed when the ListView is in "List", "SmallIcon" or "LargeIcon" views

Reviewer: Here and below: a stray "en-us".

> # ListViewAccessibleObject expected properties and behavior

Reviewer 2: Accessibility properties are not described here. You added info about views, so please rename the doc to reflect the real info.

This document describes the expected behavior of `ListViewAccessibleObject` 
when using Inspect, Narrator or AccessibilityInsights and 
expected accessibility properties of all visible items 

> of a [ListView](https://docs.microsoft.com/dotnet/api/system.windows.forms.listview).

Reviewer 2: Rework it
